### PR TITLE
Vendor Exploit Fix

### PR DESCRIPTION
**Pull request by NanakoAC**
Fixes a pretty severe exploit where spambuying from a vendor allows you to purchase a larger quantity than the vendor has. Coupled with the ability to add things, this is basically infinite item duplication. How did we not notice this before now?

I kept the spam buying, because it's pretty nifty, but added a safety check to ensure there's one remaining to buy

In addition, adds a little exception that allows bluespace technicians to take free stuff from vendors. Largely for debugging. I'm sick of having to spawn as a crewmember to do anything vendor related, this will save a bit of development time in the long run.

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -321,6 +328,10 @@
 		visible_message("<span class='info'>\The [usr] swipes \the [ID_container] through \the [src].</span>")
 	var/datum/money_account/customer_account = get_account(I.associated_account_number)
 	if (!customer_account)
+		//Allow BSTs to take stuff from vendors, for debugging and adminbus purposes
+		if (I.assignment == "Bluespace Technician")
```
Comment (Lohikar) [inline code/game/machinery/vending.dm]: This will allow people with agent IDs to get free infinite stuff.

Comment (NanakoAC) [inline code/game/machinery/vending.dm]: free but not infinite.
Don't we have some rule against impersonating a BST? I'd think someone would get bwoinked for that.

Can you think of a better way to check it?

Comment (Lohikar) [inline code/game/machinery/vending.dm]: I don't feel that this is the best way to go about solving this issue anyways.
Why not just add a loaded charge card to the BST's bag?

Comment (NanakoAC) [inline code/game/machinery/vending.dm]: i found a better way, fixed

**Pull request merged**